sunpy/image/coalignment.py

```diff
@@ -287,53 +288,26 @@ def parabolic_turning_point(y):
     return numerator / denominator
 
 
-def repair_image_nonfinite(image):
+def check_for_nonfinite_entries(layer_image, template_image):
     """
-    Return a new image in which all the nonfinite entries of the original image
-    have been replaced by the local mean.
+    Issue a warning if there is any nonfinite entry in the layer or template images.
 
     Parameters
     ----------
-    image : `numpy.ndarray`
+    layer_image : `numpy.ndarray`
+        A two-dimensional `numpy.ndarray`.
+    template_image : `numpy.ndarray`
         A two-dimensional `numpy.ndarray`.
-
-    Returns
-    -------
-    `numpy.ndarray`
-        A two-dimensional `numpy.ndarray` of the same shape as the input
-        that has all the non-finite entries replaced by a local mean. The
-        algorithm repairs one non-finite entry at every pass. At each pass,
-        the next non-finite value is replaced by the mean of its finite
-        valued nearest neighbors.
     """
-    repaired_image = deepcopy(image)
-    nx = repaired_image.shape[1]
-    ny = repaired_image.shape[0]
-    bad_index = np.where(np.logical_not(np.isfinite(repaired_image)))
-    while bad_index[0].size != 0:
-        by = bad_index[0][0]
-        bx = bad_index[1][0]
-
-        # x locations taking in to account the boundary
-        x = bx
-        if bx == 0:
-            x = 1
-        if bx == nx - 1:
-            x = nx - 2
-
-        # y locations taking in to account the boundary
-        y = by
-        if by == 0:
-            y = 1
-        if by == ny - 1:
-            y = ny - 2
-
-        # Get the sub array around the bad index, and find the local mean
-        # ignoring nans
-        subarray = repaired_image[y - 1: y + 2, x - 1: x + 2]
-        repaired_image[by, bx] = np.mean(subarray[np.isfinite(subarray)])
-        bad_index = np.where(np.logical_not(np.isfinite(repaired_image)))
-    return repaired_image
+
+    bad_index_layer = np.where(np.logical_not(np.isfinite(layer_image)))
+    bad_index_template = np.where(np.logical_not(np.isfinite(template_image)))
+
+    if bad_index_layer[0].size != 0:
```

Comment:
```suggestion
    if not np.all(np.isfinite(layer_image):
```
Since you don't actually use the output from `np.where()`, this is a simpler/cleaner pattern to use, and you can remove the `bad_index_layer = ...` line above.

Reply:
nice!